coordinator/src/dlc_protocol.rs

```diff
@@ -222,6 +223,50 @@ impl DlcProtocolExecutor {
         Ok(())
     }
 
+    /// Finishes a dlc protocol by the corresponding dlc protocol type handling.
+    pub fn finish_dlc_protocol(
+        &self,
+        protocol_id: ProtocolId,
+        trader_id: &PublicKey,
+        contract_id: &ContractId,
+        channel_id: &DlcChannelId,
+    ) -> Result<()> {
+        let mut conn = self.pool.get()?;
+        conn.transaction(|conn| {
+            let dlc_protocol = db::dlc_protocols::get_dlc_protocol(conn, protocol_id)?;
+
+            match dlc_protocol.protocol_type {
+                DlcProtocolType::Open { trade_params }
+                | DlcProtocolType::Renew { trade_params } => self.finish_trade_dlc_protocol(
+                    conn,
+                    trade_params,
+                    protocol_id,
+                    false,
+                    contract_id,
+                    channel_id,
+                ),
+                DlcProtocolType::Settle { trade_params } => self.finish_trade_dlc_protocol(
+                    conn,
+                    trade_params,
+                    protocol_id,
+                    true,
+                    contract_id,
+                    channel_id,
+                ),
+                DlcProtocolType::Rollover { .. } => self.finish_rollover_dlc_protocol(
```

Comment:
well.. Its only a technical debt because of the implication of the model name `dlc_protocol`. Originally it was called `contract_transaction` indicating that something changed the protocol, which would have fit for the rollover as well.

Naming is hard, but I think its not that bad as it is right now.